internal/file/watch/option.go

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package watch
+
+import (
+	"context"
+
+	"github.com/vdaas/vald/internal/errgroup"
+)
+
+type Option func(w *watch) error
+
+var (
+	defaultOpts = []Option{}
+)
+
+func WithErrGroup(eg errgroup.Group) Option {
+	return func(w *watch) error {
+		if eg != nil {
+			w.eg = eg
+		}
+		return nil
+	}
+}
+
+func WithDirs(dirs ...string) Option {
+	return func(w *watch) error {
+		if len(dirs) == 0 {
+			return nil
+		}
+		if w.dirs != nil {
+			w.dirs = append(w.dirs, dirs...)
+		} else {
+			w.dirs = dirs
+		}
+		return nil
+	}
+}
+
+func WithDir(dir string) Option {
```

Comment:
This function is useless because there's `WithDirs` func.

Reply:
hmm, ok